src/LibraryManager/Providers/Unpkg/UnpkgCatalog.cs

```diff
@@ -178,45 +178,51 @@ private void GetFiles(JObject fileObject, List<string> files)
             {
-                    // library name completion
-                    if (caretPosition < name.Length + 1)
-                    {
-                        IEnumerable<string> packageNames = await NpmPackageSearch.GetPackageNamesAsync(libraryNameStart, CancellationToken.None);
+                // library name completion
+                if (caretPosition < name.Length + 1)
+                {
+                    IEnumerable<string> packageNames = await NpmPackageSearch.GetPackageNamesAsync(libraryNameStart, CancellationToken.None);
 
-                        foreach (string packageName in packageNames)
+                    foreach (string packageName in packageNames)
+                    {
+                        CompletionItem completionItem = new CompletionItem
                         {
-                            CompletionItem completionItem = new CompletionItem
-                            {
-                                DisplayText = packageName,
-                                InsertionText = packageName
-                            };
+                            DisplayText = packageName,
+                            InsertionText = packageName
+                        };
 
-                            completions.Add(completionItem);
-                        }
+                        completions.Add(completionItem);
                     }
+                    
+                    completionSet.Type = CompletionType.Name;
+                }
 
-                    // library version completion
-                    else
-                    {
-                        completionSet.Start = name.Length + 1;
-                        completionSet.Length = version.Length;
+                // library version completion
+                else
+                {
+                    completionSet.Start = name.Length + 1;
+                    completionSet.Length = version.Length;
 
-                        NpmPackageInfo npmPackageInfo = await NpmPackageInfoCache.GetPackageInfoAsync(name, CancellationToken.None);
-                        foreach (SemanticVersion semVersion in npmPackageInfo.Versions)
+                    NpmPackageInfo npmPackageInfo = await NpmPackageInfoCache.GetPackageInfoAsync(name, CancellationToken.None);
+                    foreach (SemanticVersion semVersion in npmPackageInfo.Versions)
+                    {
+                        CompletionItem completionItem = new CompletionItem
                         {
-                            string itemText = name + "@" + semVersion.ToString();
-                            CompletionItem completionItem = new CompletionItem
-                            {
-                                DisplayText = itemText,
-                                InsertionText = itemText
-                            };
-
-                            completions.Add(completionItem);
-                        }
+                            DisplayText = semVersion.ToString(),
```

Comment:
>semVersion.ToString() [](start = 42, length = 21)

nit: no need to do this twice

Reply:
I see. will change.

---
In reply to: [198903355](https://github.com/aspnet/LibraryManager/pull/258#discussion_r198903355) [](ancestors = 198903355)